R/pkg/R/DataFrame.R

```diff
@@ -1138,6 +1138,11 @@ setMethod("collect",
                   if (!is.null(PRIMITIVE_TYPES[[colType]]) && colType != "binary") {
                     vec <- do.call(c, col)
                     stopifnot(class(vec) != "list")
+                    class(vec) <-
+                      if (colType == "timestamp")
+                        c("POSIXct", "POSIXt")
```

Comment:
It looks better if it won't affect other methods. I will try it. Thanks for the advice.